xlators/mount/fuse/utils/mount_glusterfs.in

```diff
@@ -35,7 +35,7 @@ _init ()
     LOG_DEBUG=DEBUG;
     LOG_TRACE=TRACE;
 
-    HOST_NAME_MAX=64;
+    HOST_NAME_MAX=255;
```

Comment:
Do we need to change at ctx.c(https://github.com/gluster/glusterfs/blob/517d75e611322dd5757d5d9231e98356273c15f9/libglusterfs/src/ctx.c#L53C5-L53C42) also. After this change maximum hostname length would be 255 chars but ctx->hostname still would be sysconf(_SC_HOST_NAME_MAX) and on my system the value is 64.

Reply:
I did some quick research to try and understand what's going on.

From what I've read, it looks like HOST_NAME_MAX  (_SC_HOST_NAME_MAX) does not refer to the FQDN, just the hostname.

The numbers do (sort of) line up with https://www.rfc-editor.org/rfc/rfc1035, which limits the FQDN to 255 bytes, but any labels to a max of 63 bytes. But looks like rfc1035 labels are 63 bytes + null and linux hostnames can be 64 bytes + null, so maybe I'm interpreting things wrong.

Pragmatically, with the glusters setup where we ran into issues, we only ran into issues with the mount commands and with this change the mounts work correctly. Checking gluster, the peers are connected via FQDNs longer than 64 characters and it's working correctly.

So I'm not sure how the MAX length in ctx plays in with all of the rest unfortunately.

Reply:
I think you are correct, we don;t need to change it.